integration/index_test.go

```diff
@@ -0,0 +1,166 @@
+			}
+		}
+		imgInfo := dockerhub(img.imgName, withPlatform(platform))
+		img.imgRef = imgInfo.ref
+		img.sociIndexDigest = optimizeImage(sh, imgInfo)
+		testImages[i] = img
+	}
+
+	return testImages
+}
+
+func TestSociIndexInfo(t *testing.T) {
+	t.Parallel()
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+	rebootContainerd(t, sh, "", "")
+
+	testImages := prepareSociIndices(t, sh)
+
+	var sociIndex soci.Index
+
+	for _, img := range testImages {
+		t.Run(img.imgName, func(t *testing.T) {
+			rawJSON := sh.O("soci", "index", "info", img.sociIndexDigest)
+			if err := json.Unmarshal(rawJSON, &sociIndex); err != nil {
+				t.Fatalf("invalid soci index from digest %s: %v", img.sociIndexDigest, rawJSON)
+			}
+			if sociIndex.MediaType != soci.OCIArtifactManifestMediaType {
+				t.Fatalf("invalid soci index media type: expected: %s, actual: %s", soci.OCIArtifactManifestMediaType, sociIndex.MediaType)
+			}
+			if sociIndex.ArtifactType != soci.SociIndexArtifactType {
+				t.Fatalf("invalid soci index artifact type: expected: %s, actual: %s", soci.SociIndexArtifactType, sociIndex.ArtifactType)
+			}
+		})
+	}
+}
+
+func TestSociIndexLs(t *testing.T) {
+	t.Parallel()
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+	rebootContainerd(t, sh, "", "")
+
+	testImages := prepareSociIndices(t, sh)
+
+	existHandlerFull := func(output string, img testImageIndex) bool {
+		// full output should have both img ref and soci index digest
+		return strings.Contains(output, img.imgRef) && strings.Contains(output, img.sociIndexDigest)
+	}
+
+	existHandlerQuiet := func(output string, img testImageIndex) bool {
+		// quiet output should only have "soci_index_digest\n"
+		return strings.Contains(output, img.sociIndexDigest+"\n")
+	}
+
+	// each test runs a soci command, filter to get expected images, and check
+	// (only) expected images exist in command output
+	tests := []struct {
+		name         string
+		command      []string
+		filter       func(img testImageIndex) bool                // return true if `img` is expected in command output
+		existHandler func(output string, img testImageIndex) bool // return true if `img` appears in `output`
+	}{
+		{
+			name:         "`soci index ls` should list all soci indices",
+			command:      []string{"soci", "index", "list"},
+			filter:       func(img testImageIndex) bool { return true },
+			existHandler: existHandlerFull,
+		},
+		{
+			name:         "`soci index ls -q` should list digests of all soci indices",
+			command:      []string{"soci", "index", "list", "-q"},
+			filter:       func(img testImageIndex) bool { return true },
+			existHandler: existHandlerQuiet,
+		},
+		{
+			name:         "`soci index ls --ref imgRef` should only list soci indices for the image",
+			command:      []string{"soci", "index", "list", "--ref", testImages[0].imgRef},
+			filter:       func(img testImageIndex) bool { return img.imgRef == testImages[0].imgRef },
+			existHandler: existHandlerFull,
+		},
```

Comment:
Can we add an explicit test here that if you do `soci index ls --ref imgRef -q` you get exactly the image ref and nothing else?

e.g. the `existHandler` should be 
```
func(output string, img testImageIndex) bool {
    return strings.Trim(output) == img.sociIndexDigest
}
```

Reply:
fixed